sklearn/utils/tests/test_multiclass.py

```diff
@@ -296,15 +298,45 @@ def test_is_multilabel():
                 for exmpl_sparse in examples_sparse:
                     assert sparse_exp == is_multilabel(
                         exmpl_sparse
-                    ), "is_multilabel(%r) should be %s" % (exmpl_sparse, sparse_exp)
+                    ), f"is_multilabel({exmpl_sparse!r}) should be {sparse_exp}"
 
             # Densify sparse examples before testing
             if issparse(example):
                 example = example.toarray()
 
             assert dense_exp == is_multilabel(
                 example
-            ), "is_multilabel(%r) should be %s" % (example, dense_exp)
+            ), f"is_multilabel({example!r}) should be {dense_exp}"
+
+
+@pytest.mark.parametrize(
+    "array_namespace, device, dtype",
+    yield_namespace_device_dtype_combinations(),
+)
+def test_is_multilabel_array_api_compliance(array_namespace, device, dtype):
+    xp, device, dtype = _array_api_for_tests(array_namespace, device, dtype)
+
+    for group, group_examples in EXAMPLES.items():
+        dense_exp = group == "multilabel-indicator"
+        for example in group_examples:
+            if not (
+                hasattr(example, "__array__")
+                and np.asarray(example).dtype.kind in "biuf"
+            ):
+                continue
+            # Densify sparse examples before testing
+            if issparse(example):
+                example = example.toarray()
```

Comment:
This line is reported as uncovered by coverage/codecov so I suspect this test to not be testing what's intended. I suspect the `continue` statement above to be reached for all sparse represented multilabel indicators.

Reply:
@ogrisel Indeed, I changed if statement to test sparse containers. It still skips some of nested sequences, and sequences that have string(object) values inside, because Array API does not support these types.